---
title: "OrkambiKids Visualizations"
author: "Virginia Rossow"
output: html_document
---


```{r setup1, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r path and packages1, include=FALSE}
library(pacman)

pacman::p_load(tidyverse, magrittr, janitor, microbiome, knitr, naniar, phyloseq, 
               textshape, Biostrings, markdown, vegan, rstatix, gdata, tableone, ggpubr, microViz)

```


```{r run tidy script, include=FALSE}

ps_orkambi_work <- readRDS("~/Documents/Forschung/Orkambi_Kids/Paper/DeepPheno_Luma-Iva/data/ps_orkambi_trimmed.rds")
ps_orkambi_work_rel <- readRDS("~/Documents/Forschung/Orkambi_Kids/Paper/DeepPheno_Luma-Iva/data/ps_orkambi_trimmed_relative.rds")

```

### Preprocessing    
```{r prepare metadata, include=FALSE}
Metadata_filt_2_1_9 <- data.frame(sample_data(ps_orkambi_work))
Metadata_filt_2_1_9$total_reads <- sample_sums(ps_orkambi_work)

metadata_unique <- Metadata_filt_2_1_9[!duplicated(Metadata_filt_2_1_9[ , c("probe_id")]), ]

#write_csv(metadata_unique, "/Users/rebecca/Documents/Forschung/Orkambi_Kids/Paper/DeepPheno_Luma-Iva/metadata_raw.csv")
# I have to add missing lung functions and sweat chlorides from id8, i do this directly in the generated csv and reload the file

metadata_complete <- read_delim("/Users/rebecca/Documents/Forschung/Orkambi_Kids/Paper/DeepPheno_Luma-Iva/data/metadata_complete.csv")
metadata_complete <- metadata_complete %>% 
  mutate(id_visit=paste(id, visit, sep="_"))

# combine with liver enzymes
liver_df <- readxl::read_excel("/Users/rebecca/Documents/Forschung/Orkambi_Kids/Paper/Liver_enzymes_all.xlsx")

liver_df <- liver_df %>% 
  mutate(id_visit=paste(id, visit, sep="_")) %>% 
  select(id_visit, GOT,GPT)

metadata_complete <- metadata_complete%>% 
  left_join(liver_df, by="id_visit")

# add ig_g response
# igg_response
# 0 = bad
# 1 = good

metadata_complete <- metadata_complete %>%
   mutate(igg_response = case_when(id == 2 | id == 3 | id == 4 ~ FALSE,
                                 TRUE ~ TRUE)) %>%
   relocate(igg_response, .before = base)

# As id 8 is often the only participant with a V8 and that ruins visualization I will create a new variable where V8 is V9
metadata_complete <- metadata_complete%>%
  mutate(visit_wo8 = case_when(visit == 8~9, TRUE~visit))

metadata_complete <- metadata_complete%>%
  mutate(baseline_ppfev1 = case_when(visit == 1 ~ pp_fev1))%>%
  group_by(id)%>%
  fill(baseline_ppfev1)%>%
  mutate(delta_ppfev1 = pp_fev1-baseline_ppfev1)%>%
  ungroup()

metadata_complete %>% 
  tabyl(id, visit)

```

```{r adjust metadata complete classes}
metadata_complete <- metadata_complete %>% 
  mutate(id=as_factor(id)) %>% 
  mutate(visit=as_factor(visit)) %>% 
  mutate(material=as_factor(material)) %>% 
  mutate(visit_wo8=as_factor(visit_wo8))

#write_rds(metadata_complete, "/Users/rebecca/Documents/Forschung/Orkambi_Kids/Paper/DeepPheno_Luma-Iva/data/metadata_combined_031123.rds")
```



```{r add updated metadata to phyloseqs}

metadata_ps <- metadata_complete %>% 
  filter(!is.na(label))

m_s <- metadata_ps %>% 
  select(probe_id, igg_response, delta_ppfev1, visit_wo8)

#write_csv(metadata_unique, "/Users/rebecca/Documents/Forschung/Orkambi_Kids/Paper/DeepPheno_Luma-Iva/metadata_unique.csv")

# add to phyloseqs
ps_orkambi_work <- ps_join(ps_orkambi_work, m_s, by="probe_id")

ps_orkambi_work_rel <- ps_join(ps_orkambi_work_rel, m_s, by="probe_id")
```

```{r preprocessing by material}

# sputum
ps_orkambi_sputum <- subset_samples(ps_orkambi_work, material == "Sputum")
ps_orkambi_sputum_rel <- subset_samples(ps_orkambi_work_rel, material == "Sputum")

# throat
ps_orkambi_throat <- subset_samples(ps_orkambi_work, material == "Throat")
ps_orkambi_throat_rel <- subset_samples(ps_orkambi_work_rel, material == "Throat")

#stool
ps_orkambi_stool <- subset_samples(ps_orkambi_work, material == "Stool")
ps_orkambi_stool_rel <- subset_samples(ps_orkambi_work_rel, material == "Stool")
```
    
# Assign colors      
```{r palette, include=FALSE}

# colours for each id
my_colors_id <- c("aquamarine3", "skyblue1", "darkgoldenrod1", "firebrick2", "chartreuse3", "sienna1", "palevioletred2", "royalblue3")
names(my_colors_id) <- levels((Metadata_filt_2_1_9$id))

ggplot(Metadata_filt_2_1_9, aes(id)) +
  geom_bar(aes(fill = id)) +
  scale_fill_manual(name = "id", values = my_colors_id) +
  theme_bw()

# colours for each visit
my_colors_visit <- c("darkorchid4", "mediumorchid2", "orchid1", "plum", "pink2", "hotpink2", "violetred1", "maroon3", "deeppink4")
names(my_colors_visit) <- levels((Metadata_filt_2_1_9$visit))

ggplot(Metadata_filt_2_1_9, aes(visit)) +
  geom_bar(aes(fill = visit)) +
  scale_fill_manual(name = "visit", values = my_colors_visit) +
  theme_bw()

# colour for each material
my_colors_material <- c("lightgreen", "steelblue1", "brown1")
names(my_colors_material) <- c("Throat", "Sputum", "Stool")

ggplot(Metadata_filt_2_1_9, aes(material)) +
  geom_bar(aes(fill = material)) +
  scale_fill_manual(name = "visit", values = my_colors_material) +
  theme_bw()

system_colors <- c("#A8465D","#AA7A38","#A5C9A5")

my_colors_id <- c("#C03728FF", "#919C4CFF", "#FD8F24FF", "#F5C04AFF", "#E68C7CFF", "#C3C377FF", "#4F5157FF", "#6F5438FF") # "#828585FF",

paletteer::paletteer_d("ggpomological::pomological_palette")     
```

### Visualizations
```{r general visualizations}

Metadata_filt_2_1_9$material <- factor(Metadata_filt_2_1_9$material, levels = c("Sputum", "Throat", "Stool"), ordered = TRUE)

# reads per material
# all materials
 Metadata_filt_2_1_9 %>%
  ggplot(., aes(x = material, y = total_reads, fill= material)) +
  geom_boxplot() +
  scale_fill_manual(name = "material", values = my_colors_material) +
  theme_bw()
 
 Metadata_filt_2_1_9 %>%
   group_by(material) %>%
   summarise(Median=as.numeric(median(total_reads)), Mean = mean(total_reads))

plot_ordination(ps_orkambi_work, ordinate(ps_orkambi_work, "MDS"), color = "material") + 
  geom_point(size = 3) +
  theme_minimal()+
  #ggtitle("Taxa by material") +
  scale_colour_manual(name = "material", values = my_colors_material) 


# with relative abundance 
TopNOTUs_rel <- names(sort(taxa_sums(ps_orkambi_work_rel), TRUE)[1:20])
ent20_rel <- prune_taxa(TopNOTUs_rel, ps_orkambi_work_rel)
plot_bar(ent20_rel, "id", fill = "genus", facet_grid = material ~ visit)+
  theme_minimal()+
  ggtitle("Most abundant genera") +
  ylab("Relative Abundance")

plot_bar(ps_orkambi_work_rel, "id", fill = "phylum", facet_grid = material ~ visit)+
  theme_minimal()+
  ggtitle("Most abundant phyla") +
  ylab("Relative Abundance")

TopNOTUs_rel_ab <- names(sort(taxa_sums(ps_orkambi_work_rel), TRUE)[1:15])
ent10_rel_ab <- prune_taxa(TopNOTUs_rel_ab, ps_orkambi_work_rel)
plot_bar(ent10_rel_ab, "material", facet_grid = ~genus) +
  theme_minimal()+
  theme(axis.text.x = element_text(size=6, angle=90)) +
  ylab("Relative Abundance") 

# qpcr data
Metadata_filt_2_1_9 %>%
 ggplot(aes(probe_id, quantity_mean))+
 geom_point(aes(color = id)) +
  theme_minimal()+
  scale_y_log10()+
  facet_grid(rows=vars(material))+
  theme(axis.text.x = element_text(size=6, angle=90))+
  labs(y = "log10 16S copies") +
  scale_colour_manual(name = "id", values = my_colors_id) 

# ppFEV1
metadata_unique %>%
  filter(pp_fev1 > 5) %>%
  ggplot(aes(x=visit, y=pp_fev1)) +
  theme_minimal()+
  geom_boxplot() +
  geom_line(aes(x = visit, group = id, color = id)) +
  geom_point(aes(x = visit, color = id)) +
  ylab("ppFEV1 (lung function)")+
  theme(text = element_text(size=16)) +
  #ggtitle("Lung Function")+
  scale_color_manual(name = "id", values = my_colors_id) 

# LCI
metadata_unique %>%
  #filter(id != 3) %>%
  ggplot(aes(x=visit, y=lci)) +
  theme_minimal()+
  geom_boxplot() +
  geom_line(aes(x = visit, group = id, color = id)) +
  geom_point(aes(x = visit, color = id)) +
  ylab("LCI")+
  theme(text = element_text(size=16)) +
  #ggtitle("Lung Function (LCI)")+
  scale_color_manual(name = "id", values = my_colors_id)

# sweatchloride
metadata_unique %>%
  filter(visit == 1 | visit == 2 | visit == 5) %>%
  ggplot(aes(x=visit, y=sweatchloride_mmol_l)) +
  theme_minimal()+
  geom_boxplot() +
  geom_line(aes(x = visit, group = id, color = id)) +
  geom_point(aes(x = visit, color = id)) +
  ylab("sweatchloride in mmol/l")+
  theme(text = element_text(size=16)) +
  #ggtitle("Lung Function (LCI)")+
  scale_color_manual(name = "id", values = my_colors_id)

# lung function: pp_fev1 between visits
Metadata_filt_2_1_9 %>%
  #filter(label != NA) %>%
  #unique(.$probe_id) %>%
  filter(pp_fev1 > 5) %>%
  ggplot(aes(x=visit, y=pp_fev1)) +
  theme_minimal()+
  geom_boxplot() +
  geom_line(aes(x = visit, group = id, color = id)) +
  geom_point(aes(x = visit, color = id)) +
  ylab("ppFEV1 (lung function)")+
  theme(text = element_text(size=16)) +
  #ggtitle("Lung Function")+
  scale_color_manual(name = "id", values = my_colors_id) 

# NMDS nicht-metrische multidimensionale Skalierung (Range werden anstelle von Werten genutzt, es wird auf Monotonie geachtet)

plot_ord_data <- ordinate(ps_orkambi_work_rel, "NMDS", "bray")
plot_ordination(ps_orkambi_work_rel, plot_ord_data, type="taxa", color = "phylum", title="Taxa") 
plot_ordination(ps_orkambi_work_rel, plot_ord_data, type="samples", color = "material", title="Samples") +
  scale_colour_manual(name = "material", values = my_colors_material)
plot_ordination(ps_orkambi_work_rel, plot_ord_data, type="split", color = "material", title="Samples and Taxa") +
  scale_colour_manual(name = "material", values = my_colors_material)


```

### Visualizations by timepoint

```{r 16s timepoints}

ps_orkambi_vx_t <- subset_samples(ps_orkambi_work, material == "Throat")
ps_orkambi_vx_t <- subset_samples(ps_orkambi_vx_t, visit_2 == "baseline" | visit_2 == "start" | visit_2 == "mid" | visit_2 == "end")
ps_orkambi_vx_st <- subset_samples(ps_orkambi_work, material == "Stool")
ps_orkambi_vx_st <- subset_samples(ps_orkambi_vx_st, visit_2 == "baseline" | visit_2 == "start" | visit_2 == "mid" | visit_2 == "end")
ps_orkambi_vx_s <- subset_samples(ps_orkambi_work, material == "Sputum")
ps_orkambi_vx_s <- subset_samples(ps_orkambi_vx_s, visit_2 == "baseline" | visit_2 == "start" | visit_2 == "mid" | visit_2 == "end")


my_comparisons <- list(c("baseline", "start"), c("baseline", "mid"), c("baseline", "end"))

sample_data(ps_orkambi_work)$material <- factor(sample_data(ps_orkambi_work)$material, levels = c("Sputum", "Throat", "Stool"), ordered = TRUE)


# general
  plot_richness(ps_orkambi_work, x = "material", color = "material", measures = c("Observed", "Shannon", "Simpson")) +
  geom_boxplot(aes(color = material)) + 
  theme_minimal()+
  geom_point(position = position_dodge(width = 0.75)) +
  scale_colour_manual(name = "material", values = my_colors_material) +
  #ggtitle("Alpha-Diversity (all materials)") +
  theme(legend.position = "none") +
  stat_compare_means(comparison=list(c("Stool", "Throat"), c("Stool", "Sputum"), c("Throat", "Sputum")), method="wilcox.test", p.adjust.method="BH", label = "p.signif")

# throat
plot_richness(ps_orkambi_vx_t, x = "visit_2", color = "visit_2", measures = c("Observed", "Shannon", "Simpson")) + 
  geom_boxplot(aes(color = visit_2)) + 
  geom_point(position = position_dodge(width = 0.75)) +
  theme_minimal()+
  #ggtitle("Alpha-Diversity (throat)") +
  #geom_line() +
  theme(legend.position = "none") +
  xlab("throat timepoints") +
  stat_compare_means(comparison=my_comparisons, method="wilcox.test", p.adjust.method="BH", label = "p.signif")

# stool
plot_richness(ps_orkambi_vx_st, x = "visit_2", color = "visit_2", measures = c("Observed", "Shannon", "Simpson")) + 
  geom_boxplot(aes(color = visit_2)) + 
  geom_point(position = position_dodge(width = 0.75)) +
  theme_minimal()+
  #ggtitle("Alpha-Diversity (stool)") +
  geom_smooth() +
  theme(legend.position = "none") +
  xlab("stool timepoints") +
  stat_compare_means(comparison=my_comparisons, method="wilcox.test", p.adjust.method="BH", label = "p.signif")

# sputum
plot_richness(ps_orkambi_vx_s, x = "visit_2", color = "visit_2", measures = c("Observed", "Shannon", "Simpson")) + 
  geom_boxplot(aes(color = visit_2)) + 
  geom_point(position = position_dodge(width = 0.75)) +
  theme_minimal()+
  #ggtitle("Alpha-Diversity (sputum)") +
  geom_smooth() +
  theme(legend.position = "none") +
  xlab("sputum timepoints") +
  stat_compare_means(comparison=my_comparisons, method="wilcox.test", p.adjust.method="BH", label = "p.signif")

```


## Visualizations devided by material (rarefied)


### Sputum

```{r sputum, warning=FALSE, fig.dim = c(8, 8)}

## general

nsamples(ps_orkambi_sputum)
ntaxa(ps_orkambi_sputum)
min(otu_table(ps_orkambi_sputum, TotalReads))
max(otu_table(ps_orkambi_sputum, TotalReads))

get_taxa_unique(ps_orkambi_sputum, "phylum")

# richness for every single sample
# er_sp <- estimate_richness(ps_orkambi_sputum, split = TRUE, measures = c("Observed", "Shannon", "Simpson"))
# er_sp
# summary(er_sp)
# IQR(er_sp$Observed)
# IQR(er_sp$Shannon)
# IQR(er_sp$Simpson)
# richness for pooled samples
estimate_richness(ps_orkambi_sputum, split = FALSE, measures = c("Observed", "Shannon", "Simpson"))

plot_bar(otu_table(ps_orkambi_sputum, TotalReads))+
  scale_y_log10() +
  ylab("Relative Abundance")
  

# Bar Plots

TopNOTUs_rel_sp <- names(sort(taxa_sums(ps_orkambi_sputum_rel), TRUE)[1:25])
ent15_rel_sp <- prune_taxa(TopNOTUs_rel_sp, ps_orkambi_sputum_rel)

TopNOTUs_rel_sp_5 <- names(sort(taxa_sums(ps_orkambi_sputum_rel), TRUE)[1:20]) # 15
ent15_rel_sp_5 <- prune_taxa(TopNOTUs_rel_sp_5, ps_orkambi_sputum_rel)

plot_bar(ps_orkambi_sputum_rel, fill="phylum") +
  facet_wrap(~id, scales="free_x", nrow=1) +
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1)) +
  ylab("Relative Abundance") +
  ggtitle("Sputum Samples on Phylum Level (by id)") # nach id

# poster plot
pp <- plot_bar(ps_orkambi_sputum_rel, fill="phylum") + 
  theme_minimal()+
  facet_wrap(~visit, scales="free_x", nrow=1) +
  theme(axis.text.x = element_blank(), legend.position = "bottom") +
  ggtitle("Sputum Samples on Phylum Level (by visit)") +
  ylab("Relative Abundance") +
  xlab("Samples")
ggsave(plot = pp, width = 6.5, height = 5, dpi = 300, filename = "sputum_samples_plot.png")

# id and visit
plot_bar(ps_orkambi_sputum_rel, "id", fill = "phylum", facet_grid = material ~ visit) + 
  theme_minimal()+
  ylab("Relative Abundance") +
  ggtitle("Sputum Samples on Phylum Level (by id and visit)")

# most abundant
plot_bar(ent15_rel_sp, fill="genus") + 
  facet_wrap(~visit, scales="free_x", nrow=1) +
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1)) +
  ggtitle("Sputum Samples on Genus Level (by visit)") +
  ylab("Relative Abundance")

# barplot like whole genome
plot_bar(ent15_rel_sp_5, fill="genus") +
  facet_wrap(~id, scales="free_x", nrow=1, labeller = "label_both") +
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1)) +
  ylab("Relative Abundance") +
  xlab("Sputum Samples")
plot_bar(ent15_rel_sp_5, fill="genus") +
  facet_wrap(~visit, scales="free_x", nrow=1, labeller = "label_both") +
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1)) +
  ylab("Relative Abundance") +
  xlab("Sputum Samples")

# most abundant
plot_bar(ent15_rel_sp, "id", fill = "genus", facet_grid = material ~ visit) + 
  theme_minimal()+
  ylab("Relative Abundance") +
  ggtitle("Sputum Samples on Genus Level (by id and visit)")


## alpha diversity

plot_richness(ps_orkambi_sputum, x = "visit", measures = c("Observed", "Shannon", "Simpson")) +
  theme_minimal()+
  geom_boxplot() +
  geom_line(aes(x = visit, group = id, color = id)) +
  geom_point(aes(x = visit, color = id)) +
  scale_colour_manual(values = my_colors_id)


# observed otus differ greatly between visits? or patients?
rich_sputum <- estimate_richness(ps_orkambi_sputum, measures=c("Observed", "Shannon", "Simpson"))
pairwise.wilcox.test(rich_sputum$Observed, sample_data(ps_orkambi_sputum)$id,  p.adjust.method = "BH") # guess that means no?
pairwise.wilcox.test(rich_sputum$Observed, sample_data(ps_orkambi_sputum)$visit,  p.adjust.method = "BH") # also no?

# v1 vs v5 vs v9
ps_orkambi_sputum_1 <- subset_samples(ps_orkambi_sputum, visit == "1")
ps_orkambi_sputum_5 <- subset_samples(ps_orkambi_sputum, visit == "5")
ps_orkambi_sputum_9 <- subset_samples(ps_orkambi_sputum, visit == "9")
ps_orkambi_sputum_159 <- merge_phyloseq(ps_orkambi_sputum_1, ps_orkambi_sputum_5, ps_orkambi_sputum_9)

rich_sputum_159 <- estimate_richness(ps_orkambi_sputum_159, measures=c("Observed", "Shannon", "Simpson"))
pairwise.wilcox.test(rich_sputum_159$Observed, sample_data(ps_orkambi_sputum_159)$visit,  p.adjust.method = "BH")
pairwise.wilcox.test(rich_sputum_159$Shannon, sample_data(ps_orkambi_sputum_159)$visit,  p.adjust.method = "BH")
pairwise.wilcox.test(rich_sputum_159$Simpson, sample_data(ps_orkambi_sputum_159)$visit,  p.adjust.method = "BH")

# er_sp_1 <- estimate_richness(ps_orkambi_sputum_1, measures=c("Observed", "Shannon", "Simpson"))
# er_sp_5 <- estimate_richness(ps_orkambi_sputum_5, measures=c("Observed", "Shannon", "Simpson"))
# er_sp_9 <- estimate_richness(ps_orkambi_sputum_9, measures=c("Observed", "Shannon", "Simpson"))
# summary(er_sp_1)
# summary(er_sp_5)
# summary(er_sp_9)
# IQR(er_sp_1$Observed)
# IQR(er_sp_1$Shannon)
# IQR(er_sp_1$Simpson)

## beta diversity

# PCoA plot using the unweighted UniFrac as distance
ordination_sp <- ordinate(ps_orkambi_sputum, method="PCoA")

plot_ordination(ps_orkambi_sputum, ordination_sp, color="visit") + 
  theme(aspect.ratio=1) +
 # ggtitle("PCoA - Visits") +
  geom_point(size = 1) +
   scale_colour_manual(values = my_colors_visit) +
  stat_ellipse() +
  theme_bw()

plot_ordination(ps_orkambi_sputum, ordination_sp, color = "id") + 
  geom_point(size = 1) +
   scale_colour_manual(values = my_colors_id) +
  theme(aspect.ratio=1) +
 # ggtitle("PCoA - Patient IDs") +
  stat_ellipse() +
  theme_bw()

plot_ord_data_sp <- ordinate(ps_orkambi_sputum_rel, "NMDS", "bray")
plot_ordination(ps_orkambi_sputum_rel, plot_ord_data_sp, type="taxa", color = "phylum", title="Taxa")
plot_ordination(ps_orkambi_sputum_rel, plot_ord_data_sp, type = "saples", color = "id", title="Samples by id") +
    scale_colour_manual(values = my_colors_id) +
  stat_ellipse()
```


### Throat

```{r throat, warning=FALSE, fig.dim = c(8, 8)}

## general
nsamples(ps_orkambi_throat)
ntaxa(ps_orkambi_throat)
min(otu_table(ps_orkambi_throat, TotalReads))
max(otu_table(ps_orkambi_throat, TotalReads))

Metadata_filt_2_1_9 %>% filter(visit == 8)

get_taxa_unique(ps_orkambi_throat, "phylum")

# richness for every single sample
# er_th <- estimate_richness(ps_orkambi_throat, split = TRUE, measures = c("Observed", "Shannon", "Simpson"))
# er_th
# summary(er_th)
# IQR(er_th$Observed)
# IQR(er_th$Shannon)
# IQR(er_th$Simpson)
# richness for pooled samples
estimate_richness(ps_orkambi_throat, split = FALSE, measures = c("Observed", "Shannon", "Simpson"))

plot_bar(otu_table(ps_orkambi_throat, TotalReads))+
  scale_y_log10() +
  ylab("Relative Abundance")

# Bar Plots

TopNOTUs_rel_th <- names(sort(taxa_sums(ps_orkambi_throat_rel), TRUE)[1:25])
ent15_rel_th <- prune_taxa(TopNOTUs_rel_th, ps_orkambi_throat_rel)

TopNOTUs_rel_th_5 <- names(sort(taxa_sums(ps_orkambi_throat_rel), TRUE)[1:15])
ent15_rel_th_5 <- prune_taxa(TopNOTUs_rel_th_5, ps_orkambi_throat_rel)
tax_table(ent15_rel_th_5)

plot_bar(ps_orkambi_throat_rel, fill="phylum") +
  facet_wrap(~id, scales="free_x", nrow=1) +
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1)) +
  ylab("Relative Abundance") +
  ggtitle("Throat Samples on Phylum Level")

plot_bar(ent15_rel_th_5, fill="genus") +
  facet_wrap(~id, scales="free_x", nrow=1) +
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1)) +
  ylab("Relative Abundance") +
  ggtitle("Throat Samples on Genus Level")

plot_bar(ps_orkambi_throat_rel, fill="phylum") + 
  theme_minimal()+
  facet_wrap(~visit, scales="free_x", nrow=1) +
  theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1)) +
  ggtitle("Throat Samples on Phylum Level (by visit)") +
  ylab("Relative Abundance")

# id and visit
plot_bar(ps_orkambi_throat_rel, "id", fill = "phylum", facet_grid = material ~ visit) + 
  theme_minimal()+
  ylab("Relative Abundance") +
  ggtitle("Sputum Samples on Phylum Level (by id and visit)")

plot_bar(ent15_rel_th, fill="genus") + 
  facet_wrap(~visit, scales="free_x", nrow=1) +
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1)) +
  ggtitle("Throat Samples on Genus Level (by visit)") +
  ylab("Relative Abundance")

# like whole genome
plot_bar(ent15_rel_th_5, fill="genus") +
  facet_wrap(~id, scales="free_x", nrow=1, labeller = "label_both") +
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1)) +
  ylab("Relative Abundance")+
  xlab("Throat Samples")
plot_bar(ent15_rel_th_5, fill="genus") +
  facet_wrap(~visit, scales="free_x", nrow=1, labeller = "label_both") +
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1)) +
  ylab("Relative Abundance") +
  xlab("Throat Samples")

plot_bar(ent15_rel_th, "id", fill = "genus", facet_grid = material ~ visit) + 
  theme_minimal()+
  ylab("Relative Abundance") +
  ggtitle("Throat Samples on Genus Level (by id)")


## alpha diversity

plot_richness(ps_orkambi_throat, x = "visit", color = "id", measures = c("Observed", "Shannon", "Simpson")) + 
  theme_minimal()+
  scale_colour_manual(values = my_colors_id)

plot_richness(ps_orkambi_throat, x = "visit", measures = c("Observed", "Shannon", "Simpson")) +
  geom_boxplot() +
  geom_jitter() +
  theme_bw()

plot_richness(ps_orkambi_throat, x = "visit", measures = c("Observed", "Shannon", "Simpson")) +
  theme_minimal()+
  geom_boxplot() +
  geom_line(aes(x = visit, group = id, color = id)) +
  geom_point(aes(x = visit, color = id)) +
  scale_colour_manual(values = my_colors_id)


# observed otus differ greatly between visits? or patients?
rich_throat <- estimate_richness(ps_orkambi_throat, measures=c("Observed", "Shannon", "Simpson"))
pairwise.wilcox.test(rich_throat$Observed, sample_data(ps_orkambi_throat)$id,  p.adjust.method = "BH") # guess that means no?
pairwise.wilcox.test(rich_throat$Observed, sample_data(ps_orkambi_throat)$visit,  p.adjust.method = "BH") # also no?

# v1 vs v5 vs v9
ps_orkambi_throat_1 <- subset_samples(ps_orkambi_throat, visit == "1")
ps_orkambi_throat_5 <- subset_samples(ps_orkambi_throat, visit == "5")
ps_orkambi_throat_9 <- subset_samples(ps_orkambi_throat, visit == "9")
ps_orkambi_throat_159 <- merge_phyloseq(ps_orkambi_throat_1, ps_orkambi_throat_5, ps_orkambi_throat_9)

rich_throat_159 <- estimate_richness(ps_orkambi_throat_159, measures=c("Observed", "Shannon", "Simpson"))
pairwise.wilcox.test(rich_throat_159$Observed, sample_data(ps_orkambi_throat_159)$visit,  p.adjust.method = "BH")
pairwise.wilcox.test(rich_throat_159$Shannon, sample_data(ps_orkambi_throat_159)$visit,  p.adjust.method = "BH")
pairwise.wilcox.test(rich_throat_159$Simpson, sample_data(ps_orkambi_throat_159)$visit,  p.adjust.method = "BH")


## beta diversity

# PCoA plot using the unweighted UniFrac as distance
ordination_th <- ordinate(ps_orkambi_throat, method="PCoA")

plot_ordination(ps_orkambi_throat, ordination_th, color="visit") + 
  theme(aspect.ratio=1) +
  #ggtitle("PCoA - Visits") +
  geom_point(size = 1) +
   scale_colour_manual(values = my_colors_visit) +
  stat_ellipse() +
  theme_bw()

plot_ordination(ps_orkambi_throat, ordination_th, color = "id") + 
  geom_point(size = 1) +
   scale_colour_manual(values = my_colors_id) +
  theme(aspect.ratio=1) +
  #ggtitle("PCoA - Patient IDs") +
  stat_ellipse() +
  theme_bw()

plot_ord_data_th <- ordinate(ps_orkambi_throat_rel, "NMDS", "bray")
plot_ordination(ps_orkambi_throat, plot_ord_data_th, type="taxa", color = "phylum", title="Taxa")
plot_ordination(ps_orkambi_throat, plot_ord_data_th, type = "samples", color = "id", title="Samples by id") +
  scale_colour_manual(values = my_colors_id) +
  stat_ellipse()

```


### Stool

```{r stool, warning=FALSE, fig.dim = c(8, 8)}

## general
nsamples(ps_orkambi_stool)
ntaxa(ps_orkambi_stool)
min(otu_table(ps_orkambi_stool, TotalReads))
max(otu_table(ps_orkambi_stool, TotalReads))

get_taxa_unique(ps_orkambi_stool, "phylum")

# richness for every single sample
# er_st <- estimate_richness(ps_orkambi_stool, split = TRUE, measures = c("Observed", "Shannon", "Simpson"))
# er_st 
# summary(er_st)
# IQR(er_st$Observed)
# IQR(er_st$Shannon)
# IQR(er_st$Simpson)

# richness for pooled samples
estimate_richness(ps_orkambi_stool, split = FALSE, measures = c("Observed", "Shannon", "Simpson"))

plot_bar(otu_table(ps_orkambi_stool, TotalReads))+
  scale_y_log10() +
  ylab("Relative Abundance")

# Bar Plots

TopNOTUs_rel_st <- names(sort(taxa_sums(ps_orkambi_stool_rel), TRUE)[1:20])
ent15_rel_st <- prune_taxa(TopNOTUs_rel_st, ps_orkambi_stool_rel)

TopNOTUs_rel_st_5 <- names(sort(taxa_sums(ps_orkambi_stool_rel), TRUE)[1:12])
ent15_rel_st_5 <- prune_taxa(TopNOTUs_rel_st_5, ps_orkambi_stool_rel)
#tax_table(ent15_rel_st_5)

plot_bar(ps_orkambi_stool_rel, fill="phylum") +
  facet_wrap(~id, scales="free_x", nrow=1) +
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1)) +
  ylab("Relative Abundance") +
  ggtitle("Stool Samples on Phylum Level")

plot_bar(ent15_rel_st_5, fill="genus") +
  facet_wrap(~id, scales="free_x", nrow=1) +
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1)) +
  ylab("Relative Abundance") #+
  #ggtitle("Stool Samples on Phylum Level")

plot_bar(ps_orkambi_stool_rel, fill="phylum") + 
  theme_minimal()+
  facet_wrap(~visit, scales="free_x", nrow=1) +
  theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1)) +
  ggtitle("Stool Samples on Phylum Level (by visit)") +
  ylab("Relative Abundance")

# id and visit
plot_bar(ps_orkambi_stool_rel, "id", fill = "phylum", facet_grid = material ~ visit) + 
  theme_minimal()+
  ylab("Relative Abundance") +
  ggtitle("Sputum Samples on Phylum Level (by id and visit)")

plot_bar(ent15_rel_st, fill="genus") + 
  facet_wrap(~visit, scales="free_x", nrow=1) +
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1)) +
  ggtitle("Stool Samples on Genus Level (by visit)") +
  ylab("Relative Abundance")

# barplot like whole genome
plot_bar(ent15_rel_st_5, fill="genus") +
  facet_wrap(~id, scales="free_x", nrow=1, labeller = "label_both") +
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1)) +
  ylab("Relative Abundance") +
  xlab("Stool Samples")
plot_bar(ent15_rel_st_5, fill="genus") +
  facet_wrap(~visit, scales="free_x", nrow=1, labeller = "label_both") +
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1)) +
  ylab("Relative Abundance") +
  xlab("Stool Samples")

plot_bar(ent15_rel_st, "id", fill = "genus", facet_grid = material ~ visit) + 
  theme_minimal()+
  ylab("Relative Abundance") +
  ggtitle("Stool Samples on Genus Level (by id)")


## alpha diversity

# it seems do get better over the visits -> investigate!
plot_richness(ps_orkambi_stool, x = "visit", color = "id", measures = c("Observed", "Shannon", "Simpson")) + 
  theme_minimal()+
  geom_line(aes(x = visit, group = id, color = id)) +
  scale_colour_manual(values = my_colors_id)

plot_richness(ps_orkambi_stool, x = "visit", measures = c("Observed", "Shannon", "Simpson")) +
  geom_boxplot() +
  geom_jitter() +
  theme_bw() 

plot_richness(ps_orkambi_stool, x = "visit", measures = c("Observed", "Shannon", "Simpson")) +
  theme_minimal()+
  geom_boxplot() +
  geom_line(aes(x = visit, group = id, color = id)) +
  geom_point(aes(x = visit, color = id)) +
  scale_colour_manual(values = my_colors_id)


# observed otus differ greatly between visits? or patients?
rich_stool <- estimate_richness(ps_orkambi_stool, measures=c("Observed", "Shannon", "Simpson"))
pairwise.wilcox.test(rich_stool$Observed, sample_data(ps_orkambi_stool)$id,  p.adjust.method = "BH") # guess that means no?
pairwise.wilcox.test(rich_stool$Observed, sample_data(ps_orkambi_stool)$visit,  p.adjust.method = "BH") # also no?

# v1 vs v5 vs v9
ps_orkambi_stool_1 <- subset_samples(ps_orkambi_stool, visit == "1")
ps_orkambi_stool_5 <- subset_samples(ps_orkambi_stool, visit == "5")
ps_orkambi_stool_9 <- subset_samples(ps_orkambi_stool, visit == "9")
ps_orkambi_stool_159 <- merge_phyloseq(ps_orkambi_stool_1, ps_orkambi_stool_5, ps_orkambi_stool_9)

rich_stool_159 <- estimate_richness(ps_orkambi_stool_159, measures=c("Observed", "Shannon", "Simpson"))
pairwise.wilcox.test(rich_stool_159$Observed, sample_data(ps_orkambi_stool_159)$visit,  p.adjust.method = "BH")
pairwise.wilcox.test(rich_stool_159$Shannon, sample_data(ps_orkambi_stool_159)$visit,  p.adjust.method = "BH")
pairwise.wilcox.test(rich_stool_159$Simpson, sample_data(ps_orkambi_stool_159)$visit,  p.adjust.method = "BH")

# er_st_1 <- estimate_richness(ps_orkambi_stool_1, measures=c("Observed", "Shannon", "Simpson"))
# er_st_5 <- estimate_richness(ps_orkambi_stool_5, measures=c("Observed", "Shannon", "Simpson"))
# er_st_9 <- estimate_richness(ps_orkambi_stool_9, measures=c("Observed", "Shannon", "Simpson"))
# summary(er_st_1)
# summary(er_st_5)
# summary(er_st_9)
# IQR(er_st_1$Observed)
# IQR(er_st_1$Shannon)
# IQR(er_st_1$Simpson)
# IQR(er_st_5$Observed)
# IQR(er_st_5$Shannon)
# IQR(er_st_5$Simpson)
# IQR(er_st_9$Observed)
# IQR(er_st_9$Shannon)
# IQR(er_st_9$Simpson)

## beta diversity

# PCoA plot using the unweighted UniFrac as distance
ordination_st <- ordinate(ps_orkambi_stool, method = "PCoA")

plot_ordination(ps_orkambi_stool, ordination_st, color = "visit") + 
  theme(aspect.ratio=1) +
  #ggtitle("PCoA - Visits (stool)") +
  geom_point(size = 1) +
   scale_colour_manual(values = my_colors_visit) +
  stat_ellipse() +
  theme_bw()

plot_ordination(ps_orkambi_stool, ordination_st, color = "id") + 
  theme(aspect.ratio=1) +
  #ggtitle("PCoA - Patient IDs (stool)") +
  geom_point(size = 1) +
  theme_minimal()+
   scale_colour_manual(values = my_colors_id) +
  stat_ellipse()

plot_ord_data_st <- ordinate(ps_orkambi_stool_rel, "NMDS", "bray")
plot_ordination(ps_orkambi_stool_rel, plot_ord_data_st, type="taxa", color = "phylum", title="Taxa")
plot_ordination(ps_orkambi_stool_rel, plot_ord_data_st, type = "sample", color = "id", title="Samples by id (stool)")+
  theme_minimal()+
  scale_colour_manual(values = my_colors_id) +
  stat_ellipse()

stressplot(plot_ord_data_st)
stressplot(plot_ord_data_sp)
#stressplot(plot_ord_data_thr)
```

# Add dominant taxa to phyloseq metadata in sputum 
```{r Add dominant taxa to phyloseq metadata}

source("orkambi_dom_taxa_functions.R")

ps_orkambi_dom_sputum_glom <- tax_glom(ps_orkambi_sputum_rel, taxrank = "genus")
ps_glom <- ps_orkambi_dom_sputum_glom
ps_full <- ps_orkambi_sputum_rel
  
  top_sputum <- find.top.taxa2(ps_glom, "genus", 1)
  top_sputum$species<- NULL
  
  rslt <- top_sputum[, "taxa"]
  dd <- matrix(unlist(rslt), nrow=1)
  colnames(dd) <- rownames(top_sputum)
  top_sputum <- t(dd)
  
  top_sputum_df <- data.frame(x1 = row.names(top_sputum), top_sputum)%>%
    mutate(dominantGenus = top_sputum)
  top_sputum_df$top_sputum <- NULL
  
  # add the dominant genus to ps
  sample_data(ps_glom)$dominant_genus <- top_sputum_df$dominantGenus
  sample_data(ps_full)$dominant_genus <- top_sputum_df$dominantGenus

```

# Add dominant taxa to phyloseq metadata in throat 
```{r Add dominant taxa to phyloseq metadata}
ps_glom_throat <-  tax_glom(ps_orkambi_throat_rel, taxrank = "genus")
ps_full_throat <- ps_orkambi_throat_rel
  
  top_throat <- find.top.taxa2(ps_glom_throat, "genus", 1)
  top_throat$species<- NULL
  
  rslt <- top_throat[, "taxa"]
  dd <- matrix(unlist(rslt), nrow=1)
  colnames(dd) <- rownames(top_throat)
  top_throat <- t(dd)
  
  top_throat_df <- data.frame(x1 = row.names(top_throat), top_throat)%>%
    mutate(dominantGenus = top_throat)
  top_throat_df$top_throat <- NULL
  
  # add the dominant genus to ps
  sample_data(ps_glom_throat)$dominant_genus <- top_throat_df$dominantGenus
  sample_data(ps_full_throat)$dominant_genus <- top_throat_df$dominantGenus
```

```{r assign comparisons}

my_comp5 <- list(c("1","2"), c("1","5"))

```

# Figure 1 systemic responses
```{r systemic responses}
my_colors_id <- c("#C03728FF", "#919C4CFF", "#FD8F24FF", "#F5C04AFF", "#E68C7CFF", "#C3C377FF", "#4F5157FF" ,"#6F5438FF")
# sweat chloride
p1 <- metadata_complete %>%
  filter(visit%in% c("1","2","5")) %>% 
  ggplot(aes(x=visit, y=sweatchloride_mmol_l)) +
  theme_minimal()+
  geom_boxplot(aes(alpha=0.7)) +
  geom_line(aes(x = visit, group = id, color = id, alpha=0.7), size=3) +
  geom_point(aes(x = visit, color = id), size=2.5) +
  ylab("Sweat chloride (mmol/l)")+
  theme(text = element_text(size=16), legend.position = "left",legend.margin = margin(50, 50, 50, 50)) +
  guides(alpha = FALSE)+
  scale_color_manual(name = "id", values = my_colors_id) #+
  #stat_compare_means(comparisons = )
p1

# Inflammation 
p2 <- metadata_complete%>%
  filter(visit!="6") %>% 
  ggplot(aes(x=visit_wo8, y= ig_g)) +
  theme_minimal()+
  geom_boxplot(aes(alpha=0.7)) +
  geom_line(aes(x = visit_wo8, group = id, color = id,alpha=0.7), size=3) +
  geom_point(aes(x = visit_wo8, color = id), size=2.5) +
  ylab("IgG (g/l)")+
  xlab("visit")+
  theme(text = element_text(size=16), legend.position = "none") +
  scale_color_manual(name = "id", values = my_colors_id)+
  scale_y_log10()

p3 <- metadata_complete%>%
  filter(visit!="6") %>% 
  ggplot(aes(x=visit_wo8, y= leukos)) +
  theme_minimal()+
  geom_boxplot(aes(alpha=0.7)) +
  geom_line(aes(x = visit_wo8, group = id, color = id,alpha=0.7), size=3) +
  geom_point(aes(x = visit_wo8, color = id), size=2.5) +
  ylab("Leucocytes /nl")+
  xlab("visit")+
  theme(text = element_text(size=16), legend.position = "none") +
  scale_color_manual(name = "id", values = my_colors_id)+
  scale_y_log10()

 # Metabolome
# load Metabolome figures from Metabolome script via loadRDS
p4 <- readRDS("/Users/rebecca/Documents/Forschung/Orkambi_Kids/Paper/DeepPheno_Luma-Iva/Ivacaftor_RP_pos.rds")
p5 <- readRDS("/Users/rebecca/Documents/Forschung/Orkambi_Kids/Paper/DeepPheno_Luma-Iva/Tryptophan_RP_neg.rds")

p4 <- p4 +
  geom_point(aes(color = as_factor(id)), size=2.5) +
  #geom_text(aes(label = as_factor(id)))+ 
  geom_boxplot(aes(alpha=0.7)) +
  ylab("Ivacaftor (MS Intensity)")+
  xlab("visit")+
  theme_minimal()+
  theme(text = element_text(size=16), legend.position = "none") +
  scale_color_manual(name = "id", values = my_colors_id)
  
p5 <- p5+
  geom_boxplot(aes(fill=NULL, alpha=0.7))+
  geom_point(aes(color = as_factor(id)), size=2.5) +
  #geom_text(aes(label = as_factor(id)))+ 
  ylab("Tryptophan (MS Intensity)")+
  xlab("visit")+
  theme_minimal()+
  theme(text = element_text(size=16), legend.position = "none") +
  scale_color_manual(name = "id", values = my_colors_id)

library(gridExtra)
# Arrange the ggplot objects in a grid
grid_arranged <- grid.arrange(
  p1 + labs(tag = "A") + theme(legend.margin = margin(25, 25, 25, 25), legend.box.spacing = unit(0, units = "pt"), legend.box.margin = margin(0, 0, 0, 0)),
  p2 + labs(tag = "B"),
  p3 + labs(tag = "C"),
  p4 + labs(tag = "D"),
  p5 + labs(tag = "E"),
  ncol = 6, nrow = 2,
  #heights = c(1.25, 1.15, 1.25, 1.35),
  layout_matrix = rbind(c(1, 1, 2, 2, 3, 3),
                        c(4, 4, 4, 5, 5, 5)
))

# Print or save the arranged plot
print(grid_arranged)

ggsave(plot=grid_arranged, "~/Documents/Forschung/Orkambi_Kids/Paper/DeepPheno_Luma-Iva/system_fig.png",dpi = 600, width = 12, height = 8)

```


# Figure 1 gastrointestinal responses
```{r gastrointestinal responses}
# BMI
p1 <- metadata_complete %>%
  ggplot(aes(x=visit, y=bmi_zscore)) +
  theme_minimal()+
  geom_boxplot(aes(alpha=0.7)) +
  geom_line(aes(x = visit, group = id, color = id, alpha=0.7), size=3) +
  geom_point(aes(x = visit, color = id), size=2.5) +
  ylab("BMI Z-score")+
   theme(text = element_text(size=16), legend.position = "none") +
  #ggtitle("Lung Function")+
  scale_color_manual(name = "id", values = my_colors_id) 

# Inflammation 
p2 <- metadata_complete%>%
  ggplot(aes(x=visit, y= calprotectin)) +
  theme_minimal()+
  geom_boxplot(aes(alpha=0.7)) +
  geom_line(aes(x = visit, group = id, color = id, alpha=0.7), size=3) +
  geom_point(aes(x = visit, color = id), size=2.5) +
  ylab("Calprotectin (µg/g) in stool")+
   theme(text = element_text(size=16), legend.position = "none") +
  #ggtitle("Lung Function")+
  scale_color_manual(name = "id", values = my_colors_id) +
  scale_y_log10()

# liver enzymes
p3 <- metadata_complete%>%
  filter(!(visit %in% c("6", "8"))) %>% 
  ggplot(aes(x=visit_wo8, y=GOT)) +
  theme_minimal() +
  geom_boxplot(aes(alpha=0.7)) +
  geom_line(aes(x = visit, group = id, color = id, alpha=0.7), size=3) +
  geom_point(aes(x = visit, color = id), size=2.5) +
  ylab("ASAT (U/I) in serum")+
  xlab("visit_wo8")+
  theme(text = element_text(size=16), legend.position = "none") +
  scale_color_manual(name = "id", values = my_colors_id)

p4 <- metadata_complete%>%
  filter(!(visit %in% c("6", "8"))) %>% 
  ggplot(aes(x=visit_wo8, y=GPT)) +
  theme_minimal()+
  geom_boxplot(aes(alpha=0.7)) +
  geom_line(aes(x = visit, group = id, color = id, alpha=0.7), size=3) +
  geom_point(aes(x = visit, color = id), size=2.5) +
  ylab("ALAT (U/I) in serum")+
  xlab("visit_wo8")+
  theme(text = element_text(size=16), legend.position = "none") +
  scale_color_manual(name = "id", values = my_colors_id)+
  scale_y_log10()

# Stool Microbiome
p5 <- plot_richness(ps_orkambi_stool, x = "visit", measures = c("Shannon")) +
  theme_minimal()+
  geom_boxplot(aes(alpha=0.7)) +
  geom_line(aes(x = visit, group = id, color = id, alpha=0.7), size=3) +
  geom_point(aes(x = visit, color = id), size=2.5) +
  scale_colour_manual(values = my_colors_id)+
  ylab("Shannon Index")+
  theme(text = element_text(size=16), legend.position = "none",strip.background = element_blank(),  # Remove the gray bar
        strip.text = element_blank() )+ # Remove the facet labels) 
  ggtitle("Stool alpha-diversity")

plot_ord_data_st <- ordinate(ps_orkambi_stool_rel, "NMDS", "bray")
p6 <- plot_ordination(ps_orkambi_stool, plot_ord_data_st)+
  geom_point(aes(color=id), size = 4.5)+
  scale_colour_manual(values = my_colors_id) +
  guides(fill = FALSE, color=FALSE, shape=FALSE)+
  theme_minimal()+
  ggtitle("Stool beta-diversity")+
  theme(text = element_text(size=20), legend.position = "none", legend.text = element_blank())

library(gridExtra)
# Arrange the ggplot objects in a grid
grid_arranged <- grid.arrange(
  p1 + labs(tag = "A") + theme(legend.margin = margin(0, 0, 0, 0), legend.box.spacing = unit(0, units = "pt"), legend.box.margin = margin(0, 0, 0, 0)),
  p2 + labs(tag = "B"),
  p3 + labs(tag = "C"),
  p4 + labs(tag = "D"),
  p5 + labs(tag = "E"),
  p6 + labs(tag = "F"),
  ncol = 6, nrow = 3,
  #heights = c(1.25, 1.15, 1.25, 1.35),
  layout_matrix = rbind(c(1, 1, 2, 2),
                        c(3, 3, 4, 4),
                        c(5, 5, 6, 6)
))

# Print or save the arranged plot
print(grid_arranged)

ggsave(plot=grid_arranged, "~/Documents/Forschung/Orkambi_Kids/Paper/DeepPheno_Luma-Iva/gastro_fig.png",dpi = 600, width = 12, height = 12)
```


# Figure 3 respiratory system
```{r}
# ppFEV1
metadata_complete$pp_fev1

p1 <- metadata_complete %>%
  filter(pp_fev1 > 5) %>%
  ggplot(aes(x=visit, y=delta_ppfev1)) +
  theme_minimal()+
  geom_boxplot(aes(alpha=0.7)) +
  geom_line(aes(x = visit, group = id, color = id, alpha=0.7), size=3) +
  geom_point(aes(x = visit, color = id), size=2.5) +
  ylab(" Delta ppFEV1 (%) from baseline")+
   theme(text = element_text(size=16), legend.position = "none") +
  #ggtitle("Lung Function")+
  scale_color_manual(name = "id", values = my_colors_id) 

# Inflammation 
p2 <- metadata_complete%>%
  filter(visit %in% c("1","2","3","4","5")) %>%
  ggplot(aes(x=visit, y= il1beta)) +
  theme_minimal()+
  geom_boxplot(aes(alpha=0.7)) +
  geom_line(aes(x = visit, group = id, color = id, alpha=0.7), size=3) +
  geom_point(aes(x = visit, color = id), size=2.5) +
  ylab("IL-1ß in sputum (pg/ml)")+
  theme(text = element_text(size=16), legend.position = "none") +
  scale_color_manual(name = "id", values = my_colors_id)+
  scale_y_log10()

p3 <- metadata_complete%>%
  filter(visit %in% c("1","2","3","4","5")) %>%
  ggplot(aes(x=visit, y= il8)) +
  theme_minimal()+
  geom_boxplot(aes(alpha=0.7)) +
  geom_line(aes(x = visit, group = id, color = id, alpha=0.7), size=3) +
  geom_point(aes(x = visit, color = id), size=2.5) +
  ylab("IL-8 in sputum (pg/ml)")+
  theme(text = element_text(size=16), legend.position = "none") +
  scale_color_manual(name = "id", values = my_colors_id)+
  scale_y_log10()

pinf <- ggarrange(p2,p3)
 
# Microbiome sputum
p4 <- plot_richness(ps_orkambi_sputum, x = "visit", measures = c("Shannon")) +
  theme_minimal()+
  geom_boxplot(aes(alpha=0.7)) +
  geom_line(aes(x = visit, group = id, color = id, alpha=0.7), size=3) +
  geom_point(aes(x = visit, color = id), size=2.5) +
  scale_colour_manual(values = my_colors_id)+
  ylab("Shannon Index")+
  theme(text = element_text(size=16), legend.position = "none",strip.background = element_blank(),  # Remove the gray bar
        strip.text = element_blank() )+ # Remove the facet labels) 
  ggtitle("Sputum alpha-diversity")

plot_ord_data_sp <- ordinate(ps_orkambi_sputum_rel, "NMDS", "bray")
p6 <- plot_ordination(ps_full, plot_ord_data_sp)+
    stat_ellipse(geom = "polygon",
               aes(fill = dominant_genus), 
               alpha = 0.15)+  
  geom_point(aes(color=id), size = 4)+
  scale_colour_manual(values = my_colors_id) +
  theme(text = element_text(size=20), legend.position = "none", legend.text = element_blank()) +
  guides(fill = FALSE, color=FALSE, shape=FALSE)+
  theme_minimal()+
  ggtitle("Sputum beta-diversity")+
  annotate("text", label = "Streptococcus",
    x = -0.2, y = 1.4, size = 6, colour = "red")+
  annotate("text", label = "Staphylococcus",
    x = 1.5, y = 0.5, size = 6, colour = "blue")


meta <- as(sample_data(ps_full), "data.frame")

summary(lmerTest::lmer(pp_fev1 ~ dominant_genus +(1|id), data = meta)) # lung function does not depend on dominantGenus

meta %>% 
  filter(dominant_genus %in% c("Staphylococcus","Streptococcus")) %>% 
  ggplot(aes(as_factor(dominant_genus), pp_fev1))+
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(aes(color = id), size=2.5) +
  scale_colour_manual(values = my_colors_id)+
  ylab("ppfev1")+
  theme(text = element_text(size=16), legend.position = "none",strip.background = element_blank(),  # Remove the gray bar
        strip.text = element_blank() )+ # Remove the facet labels) 
 stat_compare_means(comparisons = list(c("Staphylococcus","Streptococcus")), method = "wilcox.test")
  
# Throat Microbiome

p7 <- plot_richness(ps_orkambi_throat, x = "visit", measures = c("Shannon")) +
  theme_minimal()+
  geom_boxplot(aes(alpha=0.7)) +
  geom_line(aes(x = visit, group = id, color = id, alpha=0.7), size=3) +
  geom_point(aes(x = visit, color = id), size=2.5) +
  scale_colour_manual(values = my_colors_id)+
  ylab("Shannon Index")+
  theme(text = element_text(size=16), legend.position = "none",strip.background = element_blank(),  # Remove the gray bar
        strip.text = element_blank() )+ # Remove the facet labels) 
  ggtitle("Throat alpha-diversity")

plot_ord_data_th <- ordinate(ps_orkambi_throat_rel, "NMDS", "bray")
p8 <- plot_ordination(ps_full_throat, plot_ord_data_th)+
    #stat_ellipse(geom = "polygon",
               #aes(fill = dominant_genus), 
               #alpha = 0.15)+  
  geom_point(aes(color=id), size = 4)+
  scale_colour_manual(values = my_colors_id) +
  theme(text = element_text(size=20), legend.position = "none", legend.text = element_blank()) +
  guides(fill = FALSE, color=FALSE, shape=FALSE)+
  theme_minimal()+
  ggtitle("Throat beta-diversity")

library(gridExtra)
# Arrange the ggplot objects in a grid
grid_arranged <- grid.arrange(
  p1 + labs(tag = "A") + theme(legend.margin = margin(0, 0, 0, 0), legend.box.spacing = unit(0, units = "pt"), legend.box.margin = margin(0, 0, 0, 0)),
  p2 + labs(tag = "B"),
  p3 + labs(tag = "C"),
  p4 + labs(tag = "D"),
  p6 + labs(tag = "E"),
  p7 + labs(tag = "F"),
  p8 + labs(tag = "G"),
  ncol = 4, nrow = 3,
  #heights = c(1.25, 1.15, 1.25, 1.35),
  layout_matrix = rbind(c(1, 1, 2, 3),
                        c(4, 4, 5, 5),
                        c(6, 6, 7, 7)
))

# Print or save the arranged plot
print(grid_arranged)

ggsave(plot = grid_arranged,"~/Documents/Forschung/Orkambi_Kids/Paper/DeepPheno_Luma-Iva/resp_fig.png",dpi = 600, width = 12, height = 12)


ggplot(metadata_complete, aes(x = visit_wo8, y = ig_g, group = id, color = as.factor(id))) +
  geom_line() +
  geom_point() +  # Add points at data points for clarity
  labs(
    title = "IgGLevels Over Time",
    x = "Timepoint",
    y = "IgG Level",
    color = "Patient"
  ) +
  theme_minimal()
```
# Some statistics
```{r Patients characteristics}
install.packages("lmerTest")

lm <- summary(lmerTest::lmer(leukos~ visit + sex + age+ (1|id), data=metadata_complete))

  coefs <- data.frame(coef(lm))
  fdr <- p.adjust(coefs$Pr...t.., method = "fdr", n = nrow(coefs))
  
  lm_stats <- bind_cols(coefs, fdr)%>%
    mutate(p = Pr...t..)%>%
    mutate(fdr = ...6)%>%
    select(-c(Pr...t.., ...6))%>%
    rownames_to_column()%>%
    mutate(Months_after_ETI_start=rowname)%>%
    mutate(Months_after_ETI_start= case_when(Months_after_ETI_start=="(Intercept)"~"Baseline (Intercept)", Months_after_ETI_start=="visit_cal_92"~"3", Months_after_ETI_start=="visit_cal_93"~"6",Months_after_ETI_start=="visit_cal_94"~"9",Months_after_ETI_start=="visit_cal_95"~"12",Months_after_ETI_start=="visit_cal_96"~"15",Months_after_ETI_start=="visit_cal_97"~"18",Months_after_ETI_start=="visit_cal_98"~"21",Months_after_ETI_start=="visit_cal_99"~"24", Months_after_ETI_start=="sex2"~"sex", Months_after_ETI_start=="age_y"~"age_y"))%>%
    mutate(fdr_star = case_when(fdr<=0.001 ~ "***", fdr<=0.01 ~ "**", fdr<=0.05 ~ "*",fdr<=0.1 ~ ".",fdr>=0.1 ~ "ns"))%>%
    select(-rowname)%>%
    select(Months_after_ETI_start, Estimate, Std..Error, df, t.value, p, fdr, fdr_star)%>%
    mutate(p=round(p,5))%>%
    mutate(fdr=round(fdr, 5))

lm_stats

# print lm stats into data frame and add N of observations

# print lm stats into data frame and add N of observations
nobs_per_visit <- full_metadata_distinct %>%
  filter(!is.na(leukozyten_nl)) %>% 
  group_by(visit_cal_9) %>%
  summarise(N_obs = n()) %>% 
  mutate(Months_after_ETI_start= case_when(visit_cal_9=="1"~"Baseline (Intercept)", visit_cal_9=="2"~"3", visit_cal_9=="3"~"6",visit_cal_9=="4"~"9",visit_cal_9=="5"~"12",visit_cal_9=="6"~"15",visit_cal_9=="7"~"18",visit_cal_9=="8"~"21",visit_cal_9=="9"~"24"))%>% 
  mutate(Total_N_obs = sum(N_obs)) 

# Merge the two data frames
lmm_leukos <- left_join(lm_stats, nobs_per_visit, by = "Months_after_ETI_start") 
# Fill NAs in N_obs with the resulting rowsum of N_obs
lmm_leukos$N_obs <- replace_na(lmm_leukos$N_obs, lmm_leukos$Total_N_obs[1])

lmm_leukos <- lmm_leukos %>% 
  mutate(Parameter = "Leukocytes (/nl)") %>% 
  select(Parameter, Months_after_ETI_start, N_obs, Total_N_obs, everything()) %>% 
  select(-visit_cal_9)

# to make the significant asterixes printable on the plot I have to adjust the table a little
group1 <- c("(Intercept)","1","1","1","1","1","1","1","1","1","1")
  
lm_stats <-  lm_stats%>%
    bind_cols(group1)%>%
    mutate(group1=...9)%>%
  mutate(group2=case_when(Months_after_ETI_start=="3"~ "2", Months_after_ETI_start=="6"~ "3", Months_after_ETI_start=="9"~ "4",Months_after_ETI_start=="12"~ "5", Months_after_ETI_start=="15"~ "6", Months_after_ETI_start=="18"~ "7",Months_after_ETI_start=="21"~ "8", Months_after_ETI_start=="24"~ "9"))%>%
  filter(group1!="(Intercept)") %>% 
  filter(!is.na(group2))

leucos<- full_metadata_distinct%>%
    ggplot(aes(y=leukozyten_nl, x=visit_cal_9))+
     geom_boxplot(aes(fill=visit_cal_9), outlier.shape = NA, alpha=0.7)+
    geom_point(color="black", alpha=0.7, width = 0.1)+
    scale_fill_manual(values=visit_palette)+
    geom_line(aes(group=id, alpha=0.7))+
    theme_classic()+
    theme_classic()+
    ylab("Leucocytes (/nl)")+
    xlab("Months from ETI treatment start")+
    scale_y_continuous(trans=scales::pseudo_log_trans(base = 10), breaks = c(1,10,50,100,300))+
    scale_x_discrete(labels= xlabels)+
    theme(text=element_text(size = 14), legend.position = "none")+
    #guides(fill='none') +
   stat_pvalue_manual(lm_stats, label = "fdr_star",y.position = 1.4, step.increase = 0.05)
leucos

summary(lmerTest::lmer(ig_g ~ visit + (1|id), data=metadata_complete))
summary(lmerTest::lmer(leukos ~ visit + (1|id), data=metadata_complete))
summary(lmerTest::lmer(pp_fev1 ~ visit + (1|id), data=metadata_complete))
summary(lmerTest::lmer(lci~ visit + (1|id), data=metadata_complete))

view(metadata)
```

```{r session info}

sessionInfo()
```